src/psyclone/profiler.py

```diff
@@ -169,12 +174,23 @@ def __init__(self, children=None, parent=None):
         # change every time gen() is called).
         self._var_name = NameSpaceFactory().create().create_name("profile")
 
-        # Name of the region. In general at constructor time we might not
-        # have a parent subroutine or a child for the kernel, so we leave
-        # the name empty for now. The region and module names are set the
-        # first time gen() is called (and then remain unchanged).
-        self._region_name = None
+        # Name of the region. In general at constructor time we might
+        # not have a parent subroutine or a child for the kernel, so
+        # the name is left empty, unless explicitly provided by the
+        # user. The region and module names are set the first time
```

Comment:
"user. If no names are provided here then the region and module names are set..."

Reply:
Updated.

Reply:
Almost ;-) You've got "If not names are..."

Reply:
Oops.